### PR TITLE
pr_8

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

**Comment by sns-seb** [comment]
running initial analysis...

**Review by sns-seb2 — DISMISSED** [review]
SonarQube has detected 3 issues

File: src/main/java/org/sonarqube/A.java [hunk 1]
```diff
@@ -3,7 +3,11 @@
 public class A {
   String message = "foo";
 
+  // TODO fix this
   public String foo() {
+    message = null;
+    message.length();
+
     return message;
```
Comment (sns-seb2) [inline src/main/java/org/sonarqube/A.java:11]: ###AWAMlBG2b0Hum1nhz4KJ###
org.sonarqube:pr-decoration:src/main/java/org/sonarqube/A.java:BRANCH:pr_8@9
A "NullPointerException" could be thrown; "message" is nullable here.

File: src/main/java/org/sonarqube/A.java [hunk 2]
```diff
@@ -3,7 +3,11 @@
 public class A {
   String message = "foo";
 
+  // TODO fix this
   public String foo() {
+    message = null;
```
Comment (sns-seb2) [inline src/main/java/org/sonarqube/A.java]: ###AWAMlBG2b0Hum1nhz4KH###
org.sonarqube:pr-decoration:src/main/java/org/sonarqube/A.java:BRANCH:pr_8@6
Complete the task associated to this TODO comment.

File: src/main/java/org/sonarqube/A.java (hunk 1 above)
Comment (sns-seb2) [inline src/main/java/org/sonarqube/A.java:11]: ###AWAMlBG2b0Hum1nhz4KI###
org.sonarqube:pr-decoration:src/main/java/org/sonarqube/A.java:BRANCH:pr_8@9
The return value of "length" must be used.

**Comment by sns-seb** [comment]
status "2 Bugs, 1 Code Smell" added and code review added with 3 comments :ok:

**Comment by sns-seb** [comment]
analyzing again, no code change...

**Comment by sns-seb** [comment]
no new review, status still the same :ok:

**Comment by sns-seb** [comment]
removing an issue by pushing new code...

**Comment by sns-seb** [comment]
status has disappeared from Github UI
one comment outdated, the other 2 are still there

**Comment by sns-seb** [comment]
running an analysis...

**Review by sns-seb2 — DISMISSED** [review]
SonarQube has detected 2 issues

**Comment by sns-seb** [comment]
status added to PR "2 Bugs" :ok: 
preview review dismissed with message "Review applied to older or removed commit 8b7281c" :ok:
new review added, just message "SonarQube has detected 2 issues", no comment because those inserted in first review are still available on diff :ok: 
unfortunately, dismissal message and new review are not correctly ordered in the timeline :-1:

**Comment by sns-seb** [comment]
analyzing no code change...

**Comment by sns-seb** [comment]
no change to status nor review :ok:

**Comment by sns-seb** [comment]
resolving one issue as "won't fix" in SQ...

**Comment by sns-seb** [comment]
status updated to "1 Bug" :ok: 
no change to review nor comments :ok:

**Comment by sns-seb** [comment]
running an analysis...

**Comment by sns-seb** [comment]
status unchanged :ok: 
review unchanged :question:

**Comment by sns-seb** [comment]
resolving last bug in SonarQube...

**Comment by sns-seb** [comment]
status changed to "OK" :ok: 
review unchanged :ok: 
comments unchanged :ok:

**Comment by sns-seb** [comment]
running analysis...

**Review by sns-seb2 — DISMISSED** [review]

**Comment by sns-seb** [comment]
status unchanged :ok: 
previous review dismissed with message "Review dismissed because all issues are now fixed" :ok: 
new review added with status "approved" :ok:

**Comment by sns-seb** [comment]
unresolving a bug in SQ...

**Comment by sns-seb** [comment]
status updated to "1 Bug" :ok: 
review unchanged :ok: 
comments unchanged :ok:

**Comment by sns-seb** [comment]
running analysis...

**Review by sns-seb2 — DISMISSED** [review]
SonarQube has detected 1 issue

**Comment by sns-seb** [comment]
status unchanged :ok: 
preview review dismissed with message "Review dismissed because Pull Request now has new or re-opened issues" :ok:
new review added with message "SonarQube has detected 1 issue" :ok: 
order of dismissal and new review is correct :ok:

**Comment by sns-seb** [comment]
just pushed a new commit adding new content to the PR but neither adding new issue nor removing any

**Comment by sns-seb** [comment]
running analysis...

**Review by sns-seb2 — DISMISSED** [review]
SonarQube has detected 1 issue

**Comment by sns-seb** [comment]
status unchanged :ok: 
previews review dismissed with message "Review applied to older or removed commit a674216" :question:
new review added with message "SonarQube has detected 1 issue" and no comment :question:
order of dismissal and new review is incorrect :-1:

File: src/main/java/org/sonarqube/A.java (hunk 2 above)
Comment (sns-seb2) [reply src/main/java/org/sonarqube/A.java]: simulating auto reply to outdated comment

File: src/main/java/org/sonarqube/A.java (hunk 1 above)
Comment (sns-seb2) [reply src/main/java/org/sonarqube/A.java:11]: simulating auto reply to comment

Comment (sns-seb2) [reply src/main/java/org/sonarqube/A.java:11]: simulating 2nd auto reply to comment

File: src/main/java/org/sonarqube/A.java (hunk 2 above)
Comment (sns-seb2) [reply src/main/java/org/sonarqube/A.java]: simulating 2nd auto reply to outdated comment

File: src/main/java/org/sonarqube/A.java (hunk 1 above)
Comment (sns-seb) [reply src/main/java/org/sonarqube/A.java:11]: simulating non auto reply to outdated comment

Comment (sns-seb) [reply src/main/java/org/sonarqube/A.java:11]: simulating non auto reply to comment

**Review by sns-seb2 — CHANGES_REQUESTED** [review]
SonarQube has detected 1 issue

**Pull request closed without merging**